core/app/models/spree/variant.rb

```diff
@@ -302,8 +304,22 @@ def backordered?
       @backordered ||= !in_stock? && stock_items.exists?(backorderable: true)
     end
 
+    # Is this variant to be downloaded by the customer?
+    def digital?
+      digitals.present?
+    end
+
     private
 
+    # :dependent => :destroy needs to be handled manually
+    # spree does not delete variants, just marks them as deleted?
+    # optionally keep digitals around for customers who require continued access to their purchases
+    def destroy_digital
+      return if Spree::DigitalConfiguration[:keep_digitals]
+
+      digitals.map(&:destroy)
```

Comment:
```suggestion
      digitals.destroy_all
```

Reply:
@damianlegawiec If "we should always keep digitals" this whole method is not required, would you agree?